README.md

```diff
@@ -407,22 +407,19 @@ class ButtonKwargs(TypedDict):
     another: int
     maybe_var: NotRequired[int] # May be omitted
 
-class ButtonData(TypedDict):
-    variable: str
-
 class ButtonSlots(TypedDict):
-    my_slot: NotRequired[SlotFunc]
+    # Use `Slot` for slot functions.
+    my_slot: NotRequired[Slot]
+    # Use `SlotContent` when you want to allow either `Slot` instance or plain string
     another_slot: SlotContent
 
-ButtonType = Component[ButtonArgs, ButtonKwargs, ButtonSlots, ButtonData, JsData, CssData]
+ButtonType = Component[ButtonArgs, ButtonKwargs, ButtonSlots]
```

Comment:
While at it I also updated the docs on typing and validation (note: that page still has work to do). And in the README, to simplify things, I changed it to show only input validation (using only the first 3 type params).